src/features/cameras/DeleteCameraAlert.jsx

```diff
@@ -0,0 +1,134 @@
+import React, { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { red } from '@radix-ui/colors';
+import { styled } from '../../theme/stitches.config.js';
+import Button from '../../components/Button.jsx';
+import { ButtonRow } from '../../components/Form.jsx';
+import {
+  clearDeleteCameraTask,
+  deleteCamera,
+  fetchTask,
+  selectDeleteCameraLoading,
+} from '../tasks/tasksSlice.js';
+import { SimpleSpinner, SpinnerOverlay } from '../../components/Spinner.jsx';
+import { selectSelectedCamera, setSelectedCamera } from '../projects/projectsSlice.js';
+import {
+  fetchCameraImageCount,
+  selectCameraImageCount,
+  selectCameraImageCountLoading,
+  clearCameraImageCount,
+  selectDeleteCameraAlertStatus,
+  setDeleteCameraAlertStatus,
+} from './wirelessCamerasSlice.js';
+import { ASYNC_IMAGE_DELETE_BY_FILTER_LIMIT } from '../../config.js';
+import {
+  Alert,
+  AlertPortal,
+  AlertOverlay,
+  AlertContent,
+  AlertTitle,
+} from '../../components/AlertDialog.jsx';
+import { DeleteImagesProgressBar } from '../images/DeleteImagesProgressBar.jsx';
+
+const BoldText = styled('span', {
+  fontWeight: '$5',
+});
+
+const DeleteCameraAlert = () => {
+  const deleteCameraLoading = useSelector(selectDeleteCameraLoading);
+  const selectedCamera = useSelector(selectSelectedCamera);
+  const imageCount = useSelector(selectCameraImageCount);
+  const imageCountLoading = useSelector(selectCameraImageCountLoading);
+  const alertState = useSelector(selectDeleteCameraAlertStatus);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    if (imageCount === null && selectedCamera !== null && !imageCountLoading) {
+      dispatch(fetchCameraImageCount({ cameraId: selectedCamera }));
+    }
+  }, [imageCount, selectedCamera, dispatch]);
+
+  useEffect(() => {
+    if (!deleteCameraLoading.isLoading) dispatch(setDeleteCameraAlertStatus(false));
+  }, [deleteCameraLoading]);
```

Comment:
@jue-henry a couple thoughts/questions here:

- is there are reason we don't just add an `extraReducer` to the `wirelessCamerasSlice.js` that intercepts `updateDeleteCameraSuccess` / `updateDeleteCameraFailure` and closes the alert? This works & tbh I could go either way on it but just curious if you had a reason for using the `useEffect` here.
- in general, I try to follow the Redux team's recommendation to name actions based on the event that occurred, rather than as a "setter" (more on that [here](https://redux.js.org/style-guide#model-actions-as-events-not-setters)). I probably don't do a perfect job at this, but in this case I think if you stick with the `useEffect` approach here I'd change the name of the action we dispatch and variable we update in state b/c I had to dig a bit to figure out what `setDeleteCameraAlertStatus` did. "Status" is pretty vague, and a payload of `false` isn't very descriptive either. I'd use `isOpen` or something more specific.

Reply:
Good catch Natty, this piece of code can actually be deleted. We actually do catch it in the task slice and close the alert once the task has been successful. This leftover when testing a different way of closing. Also, will update the action name